### PR TITLE
8267860: Off-by-one bug when searching arrays in AlpnGreaseTest

### PR DESCRIPTION
**Pull request by driverkt**
This PR resolves: https://bugs.openjdk.org/browse/JDK-8267860

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8267860](https://bugs.openjdk.org/browse/JDK-8267860): Off-by-one bug when searching arrays in AlpnGreaseTest

### Reviewers
 * [Bradford Wetmore](https://openjdk.java.net/census#wetmore) (@bradfordwetmore - **Reviewer**) ⚠️ Review applies to [507abec6](https://git.openjdk.org/jdk/pull/9131/files/507abec6da7d18e604de74fe16254720fc4597d7)

### Contributors
 * Bradford Wetmore `<wetmore@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9131/head:pull/9131` \
`$ git checkout pull/9131`

Update a local copy of the PR: \
`$ git checkout pull/9131` \
`$ git pull https://git.openjdk.org/jdk pull/9131/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9131`

View PR using the GUI difftool: \
`$ git pr show -t 9131`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9131.diff">https://git.openjdk.org/jdk/pull/9131.diff</a>

</details>

### CODE REVIEW

File: test/jdk/sun/security/ssl/ALPN/AlpnGreaseTest.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ public class AlpnGreaseTest implements SSLContextTemplate {
             new String(greaseBytes, StandardCharsets.ISO_8859_1);
 
     private static void findGreaseInClientHello(byte[] bytes) throws Exception {
-        for (int i = 0; i < bytes.length - greaseBytes.length; i++) {
+        for (int i = 0; i < bytes.length - greaseBytes.length + 1; i++) {
```
Comment (bradfordwetmore) [inline test/jdk/sun/security/ssl/ALPN/AlpnGreaseTest.java:86]: LGTM.

**Review by bradfordwetmore — APPROVED** [review]
LGTM.

**Comment by driverkt** [comment]
Sorry for the noisy comment section... I was just trying to trigger jcheck. It looks to have been triggered.

**Comment by driverkt** [comment]
@bradfordwetmore you'll have to do /sponsor again, it looks like.

**Pull request closed without merging**